webapp/src/components/AssetTopbar/SearchBarDropdown/SearchBarDropdown.spec.tsx

```diff
@@ -0,0 +1,532 @@
+              'href',
+              locations.account(MOCKED_CREATOR.address, {
+                assetType: AssetType.ITEM,
+                section: Section.WEARABLES,
+                vendor: VendorName.DECENTRALAND,
+                page: 1,
+                sortBy: SortBy.NEWEST
+              })
+            )
+          })
+        })
+
+        it('should have the save the result in the recent searches on the click event', async () => {
+          const onSearch = jest.fn()
+          const { container } = await renderAndSelectCreatorsTab({
+            ...props,
+            onSearch
+          })
+          await waitFor(async () => {
+            const link = within(container).getByRole('link')
+            await fireEvent.click(link)
+            expect(
+              localStorage.getItem(LOCAL_STORAGE_RECENT_SEARCHES_KEY)
+            ).toBe(JSON.stringify([MOCKED_CREATOR]))
+          })
+        })
+
+        it('should not render the see all button', async () => {
+          const { queryByTestId } = await renderAndSelectCreatorsTab(props)
+          await waitFor(async () => {
+            expect(
+              queryByTestId(SEE_ALL_COLLECTIBLES_DATA_TEST_ID)
+            ).not.toBeInTheDocument()
+          })
+        })
+      })
+
+      describe('and there are no results for the search term', () => {
+        beforeEach(() => {
+          ;(catalogAPI.get as jest.Mock).mockResolvedValueOnce({
+            data: []
+          })
+        })
+        it('should render the no results text', async () => {
+          const { getByTestId } = renderSearchDropBarDropdown(props)
+
+          await waitFor(async () => {
+            expect(getByTestId(NO_RESULTS_DATA_TEST_ID)).toBeInTheDocument()
+          })
+        })
+      })
+    })
+
+    describe('and the user is searching for collections', () => {
+      const renderAndSelectCollectionsTab = async (
+        props: Partial<SearchBarDropdownProps> = {}
+      ) => {
+        const render = renderSearchDropBarDropdown(props)
+        render.debug()
+        await waitFor(() => {
+          const creatorsTab = render.getByText('Collections')
+          fireEvent.click(creatorsTab)
+        })
+        return render
+      }
+
+      beforeEach(() => {
+        localStorage.removeItem(LOCAL_STORAGE_RECENT_SEARCHES_KEY)
+      })
+
+      describe('and there are results', () => {
+        beforeEach(() => {
+          ;(builderAPI.fetchPublishedCollectionsBySearchTerm as jest.Mock).mockResolvedValue(
+            [MOCKED_COLLECTION]
+          )
+          ;(catalogAPI.get as jest.Mock).mockResolvedValue({
+            data: []
+          })
+        })
+
+        it('should render the results', async () => {
+          const render = await renderAndSelectCollectionsTab(props)
+          render.debug()
+
+          await waitFor(() => {
+            expect(render.getByText(MOCKED_COLLECTION.name)).toBeInTheDocument()
+          })
```

Comment:
If you removed the `waitFor`, did you have any problems with the tests? Their `find` counterpart (`findByText`) can be used in those cases and will avoid you from wrapping everything into a `waitFor`.
```suggestion
          expect(render.getByText(MOCKED_COLLECTION.name)).toBeInTheDocument()
```
or if you had any problems with waiting for the component to be there:
```suggestion
          expect(await render.findByText(MOCKED_COLLECTION.name)).toBeInTheDocument()
```